### PR TITLE
[Enhancement] Grant builtin storage volume usage to public role

### PR DESCRIPTION
**Pull request by TszKitLo40**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 11 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|5|00.00%|[1265, 1266, 1267, 1270, 1271]|
|:large_blue_circle:|com/starrocks/privilege/AuthorizationMgr.java|0|6|00.00%|[1750, 1752, 1754, 1755, 1757, 1759]|

File: fe/fe-core/src/main/java/com/starrocks/privilege/AuthorizationMgr.java [hunk 1]
```diff
@@ -1746,6 +1746,18 @@ public void upgradeRoleInitPrivilegeUnlock(long roleId, RolePrivilegeCollection
         LOG.info("upgrade role {}[{}]", collection.getName(), roleId);
     }
 
+    public void grantStorageVolumeUsageToPublicRole(String storageVolumeId) throws PrivilegeException {
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/privilege/AuthorizationMgr.java:1749]:
```suggestion
    public void grantDefStorageVolumeUsageToPubRole(String storageVolumeId) throws PrivilegeException {
```

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java [hunk 2]
```diff
@@ -1261,8 +1262,13 @@ private void transferToLeader() {
         if (RunMode.allowCreateLakeTable()) {
             try {
                 ((SharedDataStorageVolumeMgr) storageVolumeMgr).createOrUpdateBuiltinStorageVolume();
+                String builtinStorageVolumeId = storageVolumeMgr
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:1265]: better to encapsulate this if clause as method in StorageVolumeMgr

File: fe/fe-core/src/main/java/com/starrocks/privilege/AuthorizationMgr.java [hunk 3]
```diff
@@ -1746,6 +1746,18 @@ public void upgradeRoleInitPrivilegeUnlock(long roleId, RolePrivilegeCollection
...
+        roleWriteLock();
+        try {
+            RolePrivilegeCollection collection = getRolePrivilegeCollectionUnlocked(PrivilegeBuiltinConstants.PUBLIC_ROLE_ID,
+                    true);
+            List<PEntryObject> object = Collections.singletonList(new StorageVolumePEntryObject(storageVolumeId));
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/privilege/AuthorizationMgr.java:1754]: also plz add test case for this grant

**Review by nshangyiming — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 17 (35.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|11|00.00%|[1263, 3975, 3976, 3977, 3978, 3979, 3980, 3981, 3982, 3983, 3984]|
|:large_blue_circle:|com/starrocks/privilege/AuthorizationMgr.java|6|6|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 17 (35.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|11|00.00%|[1263, 3975, 3976, 3977, 3978, 3979, 3980, 3981, 3982, 3983, 3984]|
|:large_blue_circle:|com/starrocks/privilege/AuthorizationMgr.java|6|6|100.00%|[]|

**Review by HangyuanLiu — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**